IOPool/Input/src/PoolSource.h

```diff
@@ -80,6 +81,10 @@ namespace edm {
     bool bypassVersionCheck_;
     int const treeMaxVirtualSize_;
     RunNumber_t setRun_;
+    std::vector<RunNumber_t> setRunNumberForEachLumi_;
```

Comment:
What do you think about using the Strategy pattern to encapsulate the handling of run/lumi numbers into a specialized class where we use inheritance to separate the implementations for the different special cases?